### PR TITLE
Better CMake setup, organize includes into their own folders, use FetchContent for dependencies

### PR DESCRIPTION
**Pull request by DisabledMallis**
This deals with issue #34

### CODE REVIEW

**Review by FrederoxDev — CHANGES_REQUESTED** [review]
In Amethyst we're following the header structure of the Minecraft source.

File: AmethystAPI/CMakeLists.txt [hunk 1]
```diff
@@ -10,26 +10,117 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CONFIGURATION_TYPES "RelWithDebInfo" CACHE STRING "Build configurations" FORCE)
 set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build, options are: RelWithDebInfo" FORCE)
 
+# Fix STL structures on debug mode
+add_compile_definitions(_ITERATOR_DEBUG_LEVEL=0)
```
Comment (ZeroMemes) [inline AmethystAPI/CMakeLists.txt:14]: To fully address the runtime library incompatibility, the MSVC_RUNTIME_LIBRARY property should be set instead. While iterator debug level works for fixing some issues related to container sizes, it does not resolve the mismatch in debug/release allocator.

**Comment by Intoprelised** [comment]
hi

**Comment by DisabledMallis** [comment]
Everything *should* be good now I think...

**Comment by ZeroMemes** [comment]
😎 
```cmake
FetchContent_MakeAvailable(minhook libhat entt json glm)
```

**Comment by DisabledMallis** [comment]
> 😎
> 
> ```cmake
> FetchContent_MakeAvailable(minhook libhat entt json glm)
> ```

Is this real

**Comment by ZeroMemes** [comment]
yes

**Comment by FrederoxDev** [comment]
I'm happy with the changes to the dependencies, but still not changing around the headers. I feel pretty set on using the full paths myself

**Comment by DisabledMallis** [comment]
> I'm happy with the changes to the dependencies, but still not changing around the headers. I feel pretty set on using the full paths myself

The way it is setup right now you are able to do both, had to do that in EntityId.h

**Comment by FrederoxDev** [comment]
> I'm happy with the changes to the dependencies, but still not changing around the headers. I feel pretty set on using the full paths myself

This also includes separating them, I made the choice to merge them into one, literally a week or so ago, and have had a much better dev experience not having to search through two directory trees. If this is something that you want please make a new issue for it.

**Comment by DisabledMallis** [comment]
Okay I opened issue #36 with this PR resolving both

File: AmethystAPI/include/minecraft/src-client/common/client/renderer/screen/ScreenContext.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+#include "common/client/game/ClientInstance.h"
+#include "common/client/gui/gui/GuiData.h"
```
Comment (NguyenDuck) [inline AmethystAPI/include/minecraft/src-client/common/client/renderer/screen/ScreenContext.h:3]: Weird in here

Comment (DisabledMallis) [reply AmethystAPI/include/minecraft/src-client/common/client/renderer/screen/ScreenContext.h:3]: I dont understand

Comment (NguyenDuck) [reply AmethystAPI/include/minecraft/src-client/common/client/renderer/screen/ScreenContext.h:3]: You could be refactor them

Comment (DisabledMallis) [reply AmethystAPI/include/minecraft/src-client/common/client/renderer/screen/ScreenContext.h:3]: I suppose? I just left them in as they were

**Comment by FrederoxDev** [comment]
Closing for now, just tested it out with using Fetch content and it went up from ~3 seconds as my regular generate time to 4 and a half minutes. I might look into using git submodules, but I'm fine with having pre-compiled libs and headers right now.

**Pull request closed without merging**